lib/appExtensions.ts

```diff
@@ -0,0 +1,221 @@
+                                value
+                                localeCode
+                            }
+                        }
+                        model
+                        url
+                    }
+                }
+            }
+        }`,
+        variables: {
+            input: {
+                context: 'PANEL',
+                model: 'PRODUCTS',
+                url: '/productAppExtension/${id}',
+                label: {
+                    defaultValue: 'Sample App Default Label',
+                    locales: [
+                        {
+                            value: 'Sample App English Label',
+                            localeCode: 'en-US',
+                        },
+                        {
+                            value: 'Aplicación de muestra Española Etiqueta',
+                            localeCode: 'es-ES',
+                        },
+                    ],
+                },
+            },
+        },
+    }),
+}
+  );
+
+  return response;
+}
+
+// Creates a new App Extension that was expected on App Load
+export async function createAppExtensionOnLoad(
+  storeHash: string,
+  accessToken: string,
+  label: {
+    defaultValue: string;
+    locales: Array<{ value: string; localeCode: string }>;
+  }
+) {
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${storeHash}/graphql`,
+
+    {
+      method: "POST",
+      headers: {
+        accept: "application/json",
+        "content-type": "application/json",
+        "x-auth-token": accessToken,
+      },
+      body: JSON.stringify({
+        query: `
+            mutation AppExtension($input: CreateAppExtensionInput!) {
+                appExtension {
+                  createAppExtension(input: $input) {
+                    appExtension {
+                      id
+                      context
+                      label {
+                        defaultValue
+                        locales {
+                          value
+                          localeCode
+                        }
+                      }
+                      model
+                      url
+                    }
+                  }
+                }
+              }`,
+        variables: {
+          input: {
+            context: "PANEL",
+            model: "PRODUCTS",
+            url: "/productAppExtension/${id}",
+            label: label,
+          },
+        },
+      }),
+    }
+  );
+
+  return response;
+}
+
+// Checks for current App Extensions
+export async function checkAppExtensionsExist(
+  storeHash: string,
+  accessToken: string,
+  expectedAppExtensions: Array<{
+    defaultValue: string;
+    locales: Array<{ value: string; localeCode: string }>;
+  }>
+) {
+  // Get current App Extensions
+  const currentAppExtensions = await getAppExtensions(storeHash, accessToken);
+
+  // Get Labels of Current App Extensions
+  const currentAppExtensionLabels = await currentAppExtensions.map(
```

Comment:
this seems odd to me that we're looking up if the app extension exists by its label rather than something like an ID... when we create the app extension initially on install, what's the `response` that we get?